cedar-policy-core/src/evaluator.rs

```diff
@@ -3128,11 +3158,169 @@ pub mod test {
+                assert_eq!(advice, None);
+        });
+
+        assert_matches!(eval.interpret_inline_policy(
+            &Expr::lesseq(
+                Expr::call_extension_fn(
+                    duration_constructor.clone(),
+                    vec![Value::from("2h").into()]),
+                Value::from(1).into())),
+            Err(EvaluationError::TypeError(TypeError { expected, actual, advice, .. })) => {
+                assert_eq!(expected, nonempty![Type::Long]);
+                assert_eq!(actual, Type::Extension { name: duration_constructor.clone() });
+                assert_eq!(advice, None);
+        });
+
+        assert_matches!(eval.interpret_inline_policy(
+        &Expr::lesseq(
+            Expr::call_extension_fn(
+                duration_constructor.clone(),
+                vec![Value::from("2h").into()]),
+            Expr::call_extension_fn(
+                "decimal".parse().unwrap(),
+                vec![Value::from("2.0").into()]))),
+        Err(EvaluationError::TypeError(TypeError { expected, actual, advice, .. })) => {
+                assert_eq!(expected, nonempty![Type::Extension { name: duration_constructor.clone() }]);
+                assert_eq!(actual, Type::Extension { name: "decimal".parse().unwrap() });
+                assert_eq!(advice, None);
+        });
+
+        assert_matches!(eval.interpret_inline_policy(
+            &Expr::lesseq(
+                Expr::call_extension_fn(
+                    "decimal".parse().unwrap(),
+                    vec![Value::from("2.0").into()]),
+                Expr::call_extension_fn(
+                    duration_constructor.clone(),
+                    vec![Value::from("2h").into()]))),
+            Err(EvaluationError::TypeError(TypeError { expected, actual, advice, .. })) => {
+                assert_eq!(expected, nonempty![Type::Extension { name: duration_constructor.clone() }]);
+                assert_eq!(actual, Type::Extension { name: "decimal".parse().unwrap() });
+                assert_eq!(advice, None);
+        });
+
+        // if both sides support overloading, favor lhs
+        assert_matches!(eval.interpret_inline_policy(
+            &Expr::lesseq(
+                Expr::call_extension_fn(
+                    datetime_constructor.clone(),
+                    vec![Value::from("2023-01-23").into()]),
+                Expr::call_extension_fn(
+                    duration_constructor.clone(),
+                    vec![Value::from("2h").into()]))),
+            Err(EvaluationError::TypeError(TypeError { expected, actual, advice, .. })) => {
+                assert_eq!(expected, nonempty![Type::Extension { name: datetime_constructor.clone() }]);
+                assert_eq!(actual, Type::Extension { name: duration_constructor.clone() });
+                assert_eq!(advice, None);
+        });
+
+        // if both sides are of the same extension type without any operator overloading, remind users those that have
```

Comment:
Part of comment missing?

Reply:
Forgive my bad written English. I meant that the error message is to tell users which extension types have operator overloading.